underscore.js

```diff
@@ -1009,11 +1009,21 @@
 
   // Convert an object into a list of `[key, value]` pairs.
   _.pairs = function(obj) {
-    var keys = _.keys(obj);
-    var length = keys.length;
-    var pairs = Array(length);
-    for (var i = 0; i < length; i++) {
-      pairs[i] = [keys[i], obj[keys[i]]];
+    var pairs, i, length;
+    if (_.isMap(obj)) {
+      var entries = obj.entries();
```

Comment:
I think that's fair. I wondered that as I was coding it actually. Although, I can see value in not having to think about the details, i.e. whether it's a map or an object (which are similar enough data structures that Mozilla gives examples on when to use each). Still, I'm fine if you want to close it. Might want to have @jdalton weigh in though given that he submitted the issue in the first place.